ouroboros-consensus-protocol/src/ouroboros-consensus-protocol/Ouroboros/Consensus/Protocol/Ledger/HotKey.hs

```diff
@@ -134,19 +149,46 @@ data HotKey c m = HotKey
   -- When the key cannot evolve anymore, we poison it.
   , getInfo :: m KESInfo
   -- ^ Return 'KESInfo' of the signing key.
+  , getOCertMaybe :: m (Maybe (OCert.OCert c))
+  -- ^ Return the 'OCert' corresponding to the KES signing key, if any.
   , isPoisoned :: m Bool
-  -- ^ Return 'True' when the signing key is poisoned because it expired.
+  -- ^ Check whether a valid KES signing key exists. "Poisoned" means no
+  -- key exists; reasons for this could be:
+  -- - no signing key has been set yet
+  -- - the signing key has been explicitly erased ('forget')
+  -- - the signing key has been evolved past the end of the available
+  --   evolutions
   , sign_ ::
       forall toSign.
       (KES.Signable (KES c) toSign, HasCallStack) =>
-      toSign -> m (KES.SignedKES (KES c) toSign)
+      toSign ->
+      m (KES.SignedKES (KES c) toSign)
   -- ^ Sign the given @toSign@ with the current signing key.
   --
   -- PRECONDITION: the key is not poisoned.
   --
   -- POSTCONDITION: the signature is in normal form.
+  , forget :: m ()
```

Comment:
If we don't want the user to call this directly, perhaps we should also use `_` at the end?